File: apps/api/src/internal/redis-keys.js
export function orchestratorRootKey(repoKey) {
  return `orchestrator:state:${repoKey}:root`;
}

export function orchestratorItemsKey(repoKey) {
  return `orchestrator:state:${repoKey}:items`;
}

export function orchestratorLedgerKey(repoKey) {
  return `orchestrator:ledger:${repoKey}`;
}

export function orchestratorControlKey(repoKey) {
  return `orchestrator:control:${repoKey}`;
}

export function orchestratorIntentQueueKey({ repoKey, role }) {
  return `orchestrator:queue:intents:${String(role).toUpperCase()}:${repoKey}`;
}

File: apps/runner/redis_keys.py
from __future__ import annotations


def orchestrator_root_key(repo_key: str) -> str:
    return f"orchestrator:state:{repo_key}:root"


def orchestrator_items_key(repo_key: str) -> str:
    return f"orchestrator:state:{repo_key}:items"


def orchestrator_ledger_key(repo_key: str) -> str:
    return f"orchestrator:ledger:{repo_key}"


def orchestrator_intents_queue_key(*, role: str, repo_key: str) -> str:
    normalized_role = str(role or "").strip().upper()
    return f"orchestrator:queue:intents:{normalized_role}:{repo_key}"


def orchestrator_control_key(repo_key: str) -> str:
    return f"orchestrator:control:{repo_key}"


def telemetry_events_channel(run_id: str) -> str:
    return f"telemetry:events:{run_id}"

